drivers/interrupt_controller/intc_scr_ipic.c

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (c) 2023 Syntacore
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT scr_ipic
+
+/**
+ * @file intc_scr_ipic.c
+ * @author psk-sc
+ * @brief Syntacore Integrated Programmable Interrupt Controller (IPIC) Interface
+ *        for RISC-V processors
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/arch/cpu.h>
+#include <zephyr/init.h>
+#include <zephyr/arch/riscv/csr.h>
+#include <zephyr/device.h>
+
+#include <zephyr/sw_isr_table.h>
+#include <zephyr/drivers/interrupt_controller/intc_scr_ipic.h>
+#include <zephyr/irq.h>
+
+#define PLF_IPIC_STATIC_LINE_MAPPING CONFIG_IPIC_STATIC_LINE_MAPPING
+#define PLF_IPIC_IRQ_LN_NUM (CONFIG_EXT_IPIC_IRQ_LN_NUM)
+#define PLF_IPIC_IRQ_VEC_NUM PLF_IPIC_IRQ_LN_NUM
```

Comment:
let's leave aside the weird indentation and the random usage of parentheses but why just renaming stuff with slightly different names?